cpu/lpc2387/periph/rtc.c

```diff
@@ -56,10 +53,12 @@ void rtc_init(void)
 
     RTC_CCR = CCR_CLKSRC;                   /* Clock from external 32 kHz Osc. */
 
-    /* initialize clock with valid unix compatible values
-     * If RTC_YEAR contains an value larger unix time_t we must reset. */
-    if (RTC_YEAR > 2037) {
-        _rtc_set(0);
+    /* Initialize clock to a a sane and predictable default
+     * after cold boot or external reset.
+     */
+    if ((RSIR == RSIR_POR) || (RSIR == (RSIR_POR | RSIR_EXTR))) {
```

Comment:
```suggestion
    if (RSIR & RSIR_POR) {
```

This works on the MSB-A2 reliably to detect external reset and cold boot.

Reply:
I'm afraid it then will also reset the clock when you do a simple `reboot`.